c7n/resources/cloudfront.py

```diff
@@ -21,6 +21,21 @@ class DescribeDistribution(DescribeSource):
     def augment(self, resources):
         return universal_augment(self.manager, resources)
 
+    def get_resources(self, ids, cache=True):
+        results = []
+        distribution_ids = []
+        for id in ids:
+            # if we get cloudfront distribution arn, we pick distribution id
+            if 'arn:aws:cloudfront' in id:
```

Comment:
I think this explicit ARN segment will fail outside the standard AWS partition (wouldn't match govcloud or china ARNs). The general flow here seems reasonable, but maybe there's a more flexible option like:
* Checking for `id.startswith('arn:')` (this _seems_ sufficient but I may be oversimplifying)
* Using splits or regexes to catch `arn:<something>:cloudfront:<something>` patterns

Reply:
we should probably grab the partition from the current region to be partition agnostic, aws gov cloud's don't have cloud front afaik, but china partition does.

Reply:
Updated condition to id.startswith('arn:') which should work for Standard AWS partition and China ARNs